addon/components/x-select.js

```diff
@@ -72,38 +72,99 @@ export default Ember.Component.extend({
   tabindex: 0,
 
   /**
+   * Function for the `on-blur` action
+   *
+   * @property on-blur
+   * @type Function
+   */
+  "on-blur": Ember.K,
+
+  /**
+   * Function for the `on-click` action
+   *
+   * @property on-click
+   * @type Function
+   */
+  "on-click": Ember.K,
+
+  /**
+   * Function for the `on-change` action
+   *
+   * @property on-change
+   * @type Function
+   */
+  "on-change": Ember.K,
+
+  /**
+   * Function for the `on-focus-out` action
+   *
+   * @property on-focus-out
+   * @type Function
+   */
+  "on-focus-out": Ember.K,
+
+  /**
+   * Object that holds the current state of the component
+   *
+   * @property componentState
+   * @type Object
+   * @return {Object} current state
+   */
+  componentState: Ember.computed('disabled', 'multiple', 'required', 'autofocus', function() {
+    return {
+      isDisabled: this.get('disabled'),
+      isMultiple: this.get('multiple'),
+      isAutoFocus: this.get('autofocus'),
+      isRequired: this.get('required')
+    };
+  }),
+
+  /**
+   * Function that calls an action and sends the proper arguments.
+   *
+   * @method _handleAction
+   * @type Function
+   * @param {String} action - string name of the action to invoke
+   * @param {String|Object} value - current value of the component
+   * @param {Object} event - jQuery event from the current action
+   */
+  _handleAction(action, value, event) {
+    this.get(action)(value, event, this.get('componentState'));
+  },
+
+  /**
    * When the select DOM event fires on the element, trigger the
    * component's action with the current value.
    */
   change(event) {
     let nextValue = this._getValue();
 
-    this.sendAction('action', nextValue, this);
-    this.sendAction('onchange', this, nextValue, event);
+    this.sendAction('action', nextValue, event, this.get('componentState'));
+    this._handleAction('on-change', nextValue, event);
```

Comment:
Does this dispatch correctly? I'm concerned that bubbling an action with `sendAction` AND calling the closure action could just dispatch the action twice? Is `sendAction` a no-op if a closure action is passed in or something?